[UI] Make categories and filters be clickable, not hoverable

veryprofessionaldodo opened the pull request:
Noticed that the behavior of the filter and categories was a bit cumbersome on a gamepad, since it automatically expanded when hovered, and navigating was hit or miss. Since this behavior was used in several places, I made a small component to handle the expansion logic, with custom gamepad rules for entering and exiting the children. 

I am already working on adding this component to simplify some CSS as well for the DotsMenu, but I wanted to confirm with you whether you like this approach. 

The only thing I couldn't fix (more complex than I thought :( ) was clicking with a mouse on the new Dropdown component. Before clicking only made the auto-expandable container sticky, but it didn't collapse. Now it doesn't collapse as well when clicking, but it does collapse when the container loses focus. 

I also simplified some CSS for the header that used a grid, but noticed it was a bit less flexible when testing on smaller screens. Also highlighted how it works in the video below:

<details>
Before:

https://github.com/user-attachments/assets/3cc1decd-94fd-45c1-928e-feab4f0ce11d

After:

https://github.com/user-attachments/assets/19ee964e-915e-46d1-af2b-c060b373f32f

</details> 

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

arielj commented:
> @arielj do you remember if there is a reason for it to be the way it is? if related with the joystick navigation or the accessibility?

well the reason to show the content on hover was so you don't have to click it to show it haha

I can see though how it could be kinda annoying if it's focused unintentionally when using a controller though so I think it's fine to disable that for a controller (I haven't checked the code, but I'd keep the `show it on hover` when using a mouse though)

veryprofessionaldodo commented:
@arielj regarding the comments:
- the popUpOnHover is not used, but since it is a new component, we might want to maintain consistency with existing behavior, so we could just add the prop to the <Dropdown> in the category and library filters and it would be similar to how it was before. That isn't working is strange, the videos I did in the comment above show how it was working. I think you might need to refresh the page when changing the popUpOnHover prop though
- yup, they were change to "tab" and "shift+tab" because as you could see in the "before" video on the description, navigating with gamepad (and keyboard) wasn't really accessible, since focus was stolen by other elements in the way of the dropdown. This way we can ensure that all the options are fully accessible, even if they have to pass the "only" in the toggle

Let me know if there's anything else I can do for this MR :)

arielj commented:
> * yup, they were change to "tab" and "shift+tab" because as you could see in the "before" video on the description, navigating with gamepad (and keyboard) wasn't really accessible, since focus was stolen by other elements in the way of the dropdown. This way we can ensure that all the options are fully accessible, even if they have to pass the "only" in the toggle

maybe the `tab-index` attributes can be changed so the tab doesn't go to `only`

arielj commented:
> @arielj In that case it might be a bug, I'll double check the behavior and report back 👀
> 
> I don't think setting the 'tab-index' attribute is a good idea here, because in that case we would be disabling accessibility of that feature to gamepad users and keyboard navigation. I did do [another MR](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/5018) for this, but that is it's own separate conversation :)

you could still move to the right to focus the `only`, the tab index would only affect the order the tab moves up and down

veryprofessionaldodo commented:
Doubly strange, I also can't replicate this on Linux, just tested now. This is on EndeavorOS KDE, so I'm very confused


https://github.com/user-attachments/assets/5e5d1c2d-afe4-4bc7-b18e-f046cb63ca56

veryprofessionaldodo commented:
@arielj I apologise for the late reply, but I bring good news! I added in the popupOnHover to replicate the previous behavior of hovering, and I also removed the changes to the gamepad acting as a tab. This was a workaround and I wasn't really proud of the solution I gave, so I managed to fix the underlying issue with focus is escaping with navigation, so now it works for keyboards as well! Here's a video of it in action, now using just the keyboard for navigation: 

https://github.com/user-attachments/assets/a7842fe0-97eb-41ef-a7d9-a8d94f7d27f2

However, I didn't get to replicate the issue you found with the "only" appearing twice, but that might've been fixed regardless. Could've been a change with the hover behavior, and I did change that part in the CSS

flavioislima commented:
@arielj added this one to the next milestone so would be good to try again when you have the time.

arielj commented:
> Let me know if there's anything else I can do to help testing this!

I see the PR has conflicts, can you check that?

veryprofessionaldodo commented:
@arielj Fixed now!
